cmd/bbolt/command_surgery_meta.go

```diff
@@ -0,0 +1,317 @@
+	}
+
+	return nil
+}
+
+type surgeryMetaUpdateOptions struct {
+	surgeryBaseOptions
+	fields   []string
+	pageSize uint32
+}
+
+var allowedMetaUpdateFields = map[string]struct{}{
+	"pageSize": {},
+	"root":     {},
+	"freelist": {},
+	"pgid":     {},
+	"txid":     {},
+}
+
+// AddFlags sets the flags for `meta update` command.
+// Example: --fields root:16,freelist:8 --fields pgid:128 --fields txid:1234
+// Result: []string{"root:16", "freelist:8", "pgid:128", "txid:1234"}
+func (o *surgeryMetaUpdateOptions) AddFlags(fs *pflag.FlagSet) {
+	o.surgeryBaseOptions.AddFlags(fs)
+	fs.StringSliceVarP(&o.fields, "fields", "", []string{}, "comma separated list of fields (supported fields: pageSize, root, freelist, pgid and txid) to be updated, and each item is a colon-separated key-value pair")
+	fs.Uint32VarP(&o.pageSize, "page-size", "", o.pageSize, "page size (used to locate the second meta page, use the os default page size if zero")
+}
+
+func (o *surgeryMetaUpdateOptions) Validate() error {
+	if err := o.surgeryBaseOptions.Validate(); err != nil {
+		return err
+	}
+
+	for _, field := range o.fields {
+		kv := strings.Split(field, ":")
+		if len(kv) != 2 {
+			return fmt.Errorf("invalid key-value pair: %s", field)
+		}
+
+		if _, ok := allowedMetaUpdateFields[kv[0]]; !ok {
+			return fmt.Errorf("field %q isn't allowed to be updated", kv[0])
+		}
+
+		if _, err := strconv.ParseUint(kv[1], 10, 64); err != nil {
+			return fmt.Errorf("invalid value %q for field %q", kv[1], kv[0])
+		}
+	}
+
+	return nil
+}
+
+func newSurgeryMetaUpdateCommand() *cobra.Command {
+	var o surgeryMetaUpdateOptions
+	metaUpdateCmd := &cobra.Command{
+		Use:   "update <bbolt-file> [options]",
+		Short: "Update fields in meta pages",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("db file path not provided")
+			}
+			if len(args) > 1 {
+				return errors.New("too many arguments")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			return surgeryMetaUpdateFunc(args[0], o)
+		},
+	}
+	o.AddFlags(metaUpdateCmd.Flags())
+	return metaUpdateCmd
+}
+
+func surgeryMetaUpdateFunc(srcDBPath string, cfg surgeryMetaUpdateOptions) error {
+	if _, err := checkSourceDBPath(srcDBPath); err != nil {
+		return err
+	}
+
+	if err := common.CopyFile(srcDBPath, cfg.outputDBFilePath); err != nil {
+		return fmt.Errorf("[meta update] copy file failed: %w", err)
+	}
+
+	pageSize := cfg.pageSize
+	if pageSize == 0 {
+		pageSize = uint32(common.DefaultPageSize)
+	}
+
+	// update meta page 0
+	m0, buf0, err := ReadMetaPageAt(cfg.outputDBFilePath, 0, pageSize)
+	if err != nil {
+		return fmt.Errorf("read the first meta page failed: %w", err)
+	}
+	m0Changed := updateMetaField(m0, cfg.fields)
+	if m0Changed {
+		if err := writeMetaPageAt(cfg.outputDBFilePath, buf0, 0, pageSize); err != nil {
```

Comment:
Just thinking about what if the page size has been updated. Should we still use the old pageSize here?

Reply:
And should we update the page size here? If I understand correctly, the layout has been fixed with existing page size. The surgery might have accident if we change page size😂

Reply:
The `surgery meta` command is only supposed to be used by advanced users. We don't trust any data (including the pageSize) persisted in the db file if we use the `surgery meta` command; instead we only trust the values provided by the advanced users.

Reply:
Good point.